Microsoft.HealthVault.Fhir/Transformers/ObservationToHealthVault.cs

```diff
@@ -32,39 +35,97 @@ public static class ObservationToHealthVault
         /// <returns>The HealthVault thing</returns>
         public static ThingBase ToHealthVault(this Observation observation)
         {
-            // ToDo: detect the type from the codeable values
-            return observation.ToWeight();
+            return observation.ToHealthVault(VocabToHealthVaultHelper.DetectHealthVaultTypeFromObservation(observation));
         }
 
-        private static Weight ToWeight(this Observation observation)
+        private static ThingBase ToHealthVault(this Observation observation, Type type)
         {
-            var weight = new Weight();
-            var value = observation.Value as Quantity;
+            if (type == typeof(Weight))
+            {
+                return observation.ToWeight();
+            }
```

Comment:
We will figure out a better approach once we add more types. Yes, the idea is to create a registry similar to what we are doing with ToObservation right now. 

---
In reply to: [124305682](https://github.com/Microsoft/healthvault-fhir-library/pull/3#discussion_r124305682) [](ancestors = 124305682,124273929)